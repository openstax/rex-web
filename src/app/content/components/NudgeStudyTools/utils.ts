import { HTMLElement } from '@openstax/types/lib.dom';
import * as Cookies from 'js-cookie';
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useDebouncedWindowSize, useOnScrollTopOffset } from '../../../reactUtils';
import { assertDocument, remsToPx } from '../../../utils';
import { page as pageSelector } from '../../selectors';
import { hasStudyGuides } from '../../studyGuides/selectors';
import { Page } from '../../types';
import {
  arrowDesktopHeight,
  arrowRightMargin,
  arrowTopMargin,
  closeButtonDistanceFromContent,
  closeButtonSize,
  contentMarginLeft,
  contentMarginTop,
  contentWidth,
  cookieNudge,
  daysUntilCookieExpires,
  mobileNudgeStudyToolsTargetId,
  mobileSpotlightPadding,
  nudgeStudyToolsMinPageLimit,
  nudgeStudyToolsShowLimit,
  nudgeStudyToolsTargetId,
  timeIntervalBetweenShowingNudgeInMs,
} from './constants';

interface Positions {
  arrowLeft: number;
  arrowTopOffset: number;
  closeButtonLeft: number;
  closeButtonTopOffset: number;
  contentWrapperLeft: number;
  contentWrapperTopOffset: number;
  spotlightHeight: number;
  spotlightLeftOffset: number;
  spotlightTopOffset: number;
  spotlightWidth: number;
}

export const getPositions = (target: HTMLElement, isMobile: boolean): Positions => {
  const padding = remsToPx(mobileSpotlightPadding);
  const { top, left, height, width } = getBoundingRectOfNudgeTarget(target);
  const spotlightLeftOffset = left - (isMobile ? padding : 0);
  const spotlightWidth = width + (isMobile ? padding * 2 : 0);
  const arrowTopOffset = top + height + remsToPx(arrowTopMargin);
  // left edge of arrow image should be on the middle of the spotlight (adjusted for a margin)
  const arrowLeft = spotlightLeftOffset + (spotlightWidth / 2) - remsToPx(arrowRightMargin);

  const contentWrapperTopOffset = arrowTopOffset
    + remsToPx(arrowDesktopHeight)
    + remsToPx(contentMarginTop);
  const contentWrapperLeft = spotlightLeftOffset + spotlightWidth + remsToPx(contentMarginLeft);
  const closeButtonLeft = contentWrapperLeft + remsToPx(contentWidth) + remsToPx(closeButtonDistanceFromContent);
  const closeButtonTopOffset = contentWrapperTopOffset
    - remsToPx(closeButtonSize)
    - remsToPx(closeButtonDistanceFromContent);

  return {
    arrowLeft,
    arrowTopOffset,
    closeButtonLeft,
    closeButtonTopOffset,
    contentWrapperLeft,
    contentWrapperTopOffset,
    spotlightHeight: height,
    spotlightLeftOffset,
    spotlightTopOffset: top,
    spotlightWidth,
  };
};

// Target may have display set to `contents` so we are calculatings rect for children of passed target.
const getBoundingRectOfNudgeTarget = (target: HTMLElement) => {
  const { top, bottom, left, right } = Array.from(target.children).reduce((rects, child) => {
    const rect = child.getBoundingClientRect();
    rects.top = Math.min(rects.top, rect.top);
    rects.bottom = Math.max(rects.bottom, rect.bottom);
    rects.left = Math.min(rects.left, rect.left);
    rects.right = Math.max(rects.right, rect.right);
    return rects;
    // Starting values depends on if we checks for Math.max or Math.min
  }, { top: 9999, bottom: 0, left: 9999, right: 0 });

  return {
    height: bottom - top,
    left,
    right,
    top,
    width: right - left,
  };
};

export const usePositions = (isMobile: boolean) => {
  const [windowWidth] = useDebouncedWindowSize();
  const scrollTopOffset = useOnScrollTopOffset();
  const [positions, setPositions] = React.useState<Positions | null>(null);
  const studyGuides = useSelector(hasStudyGuides);

  React.useEffect(() => {
    const document = assertDocument();
    const target = document.querySelector(
      `#${isMobile ? mobileNudgeStudyToolsTargetId : nudgeStudyToolsTargetId}`
    ) as HTMLElement | null;
    if (target) {
      // Make sure that we calculate positions with body overflow set to hidden
      // because it causes scrollbar to hide which results in different positions.
      const prevOverflow = document.body.style.overflow;
      document.body.style.overflow = 'hidden';
      setPositions(getPositions(target, isMobile));

      // Resets to the value from before calculations. We want this style change to be handled
      // directly in the component.
      document.body.style.overflow = prevOverflow;
    }
    return () => setPositions(null);
  }, [scrollTopOffset, studyGuides, windowWidth, isMobile]);

  return positions;
};

export const getCounterCookie = () => {
  return Number(Cookies.get(cookieNudge.counter) || 0);
};

export const getDateCookie = () => {
  const lastShownDate = Cookies.get(cookieNudge.date);
  return lastShownDate ? new Date(lastShownDate) : undefined;
};

export const passedTimeInterval = () => {
  const lastShownDate = getDateCookie();
  return !lastShownDate
    ? true
    : (Date.now() - lastShownDate.getTime()) > timeIntervalBetweenShowingNudgeInMs
  ;
};

export const getPageCounterCookie = () => {
  return Number(Cookies.get(cookieNudge.pageCounter) || 0);
};

export const incrementPageCounterCookie = () => {
  const counter = getPageCounterCookie();
  const newValue = counter + 1;
  Cookies.set(cookieNudge.pageCounter, newValue.toString(), {expires: daysUntilCookieExpires});
  return newValue;
};

export const shouldDisplayNudgeStudyTools = (): boolean => {
  const counter = getCounterCookie();
  const numberOfPagesOpenedByUser = getPageCounterCookie();

  return counter < nudgeStudyToolsShowLimit
    && passedTimeInterval()
    && (numberOfPagesOpenedByUser >= nudgeStudyToolsMinPageLimit);
};

// Set required cookies and reset opened page counter
export const setNudgeStudyToolsCookies = () => {
  const now = new Date();
  const counter = getCounterCookie();
  Cookies.set(cookieNudge.counter, (counter + 1).toString(), {expires: daysUntilCookieExpires});
  Cookies.set(cookieNudge.date, now.toString(), {expires: daysUntilCookieExpires});
  Cookies.remove(cookieNudge.pageCounter);
};

export const useIncrementPageCounter = () => {
  const page = useSelector(pageSelector);
  const [counter, updateCounter] = useState<number>(getPageCounterCookie());
  const pageRef = React.useRef<Page>();

  React.useEffect(() => {
    if (page && page !== pageRef.current && counter <= nudgeStudyToolsMinPageLimit) {
      pageRef.current = page;
      updateCounter(incrementPageCounterCookie());
    }
  }, [page, updateCounter, counter]);

  return counter;
};
