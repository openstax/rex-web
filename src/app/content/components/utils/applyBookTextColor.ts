import { css } from 'styled-components/macro';
import theme from '../../../theme';
import { BookWithOSWebData } from '../../types';

export const applyBookTextColor = (props: {colorSchema: BookWithOSWebData['theme'] | undefined }) =>
  props.colorSchema && css`
    color: ${theme.color.primary[props.colorSchema].foreground};
  `;
