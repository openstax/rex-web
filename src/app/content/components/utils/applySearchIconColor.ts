import { css } from 'styled-components/macro';
import theme from '../../../theme';
import { toolbarIconColor } from '../constants';

const primaryColors = theme.color.primary;

export const applySearchIconColor = (colorSchema: keyof typeof primaryColors | null) =>
  colorSchema ? css`
    color: ${primaryColors[colorSchema].foreground};

    :hover,
    :focus {
        color: ${primaryColors[colorSchema].foreground};
        transition: color 200ms;

        ::before {
            content:"";
            display: block;
            height: 100%;
            position: absolute;
            top: 0;
            right: 0;
            min-width: 45px;
            background-color: rgba(0, 0, 0, 0.06)
            ;
        }
    }
  ` : css`
  color: ${toolbarIconColor.base};
`;
