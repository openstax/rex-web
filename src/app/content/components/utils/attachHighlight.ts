import Highlighter, { Highlight, SerializedHighlight } from '@openstax/highlighter';
import Sentry, { Severity } from '../../../../helpers/Sentry';

const attachHighlight = <T extends Highlight | SerializedHighlight>(
  highlight: T,
  highlighter: Highlighter,
  errorMsg?: (highlight: T) => string
): Highlight | undefined => {
  let attachedHighlight: Highlight | undefined;

  try {
    highlighter.highlight(highlight);
    attachedHighlight = highlighter.getHighlight(highlight.id);
  } catch (e) {
    Sentry.captureException(e);
  }

  if (!attachedHighlight || !attachedHighlight.isAttached()) {
    Sentry.captureException(
      new Error(errorMsg ?  errorMsg(highlight) : `Highlight with id: ${highlight.id} has not been attached.`),
      Severity.Warning
    );
    attachedHighlight = undefined;
  }

  return attachedHighlight;
};

export default attachHighlight;
