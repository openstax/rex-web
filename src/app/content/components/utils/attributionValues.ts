import { Book } from '../../types';

// date is initialized as UTC, conversion to local time can change the date.
// this compensates
export function compensateForUTC(date: Date): void {
  date.setMinutes(date.getMinutes() + date.getTimezoneOffset());
}

export function getPublishDate(book: Book): Date | null {
  if (!('publish_date' in book)) {
    return null;
  }
  const date = new Date(book.publish_date);

  compensateForUTC(date);
  return date;
}

export function getAuthors(book: Book) {
  if (!('authors' in book)) {
    return [];
  }
  const seniorAuthors = book.authors.filter(
    author => author.value.senior_author
  );

  return seniorAuthors.length > 0 ? seniorAuthors : book.authors.slice(0, 2);
}

export const bookIdsWithSpecialAttributionText: {
  [key: string]: {
    copyrightHolder?: string;
    originalMaterialLink?: null | string;
  };
} = {
  '1b4ee0ce-ee89-44fa-a5e7-a0db9f0c94b1': {
    copyrightHolder: 'The Michelson 20MM Foundation',
  },
  '394a1101-fd8f-4875-84fa-55f15b06ba66': {
    copyrightHolder: 'Texas Education Agency (TEA)',
    originalMaterialLink: 'https://www.texasgateway.org/book/tea-statistics',
  },
  'cce64fde-f448-43b8-ae88-27705cceb0da': {
    copyrightHolder: 'Texas Education Agency (TEA)',
    originalMaterialLink: 'https://www.texasgateway.org/book/tea-physics',
  },
};

export function attributionValues(book: Book) {
  const bookPublishDate = getPublishDate(book);
  const authorsToDisplay = getAuthors(book);
  const locale = book.language;

  return {
    bookTitle: book.title,
    publisher: locale === 'pl' ? 'OpenStax Poland' : 'OpenStax',
    language: new Intl.DisplayNames([locale], { type: 'language' }).of(locale) as string,
    bookPublishDate,
    authorsToDisplay,
  };
}
