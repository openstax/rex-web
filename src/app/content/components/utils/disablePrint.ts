import { css } from 'styled-components';

export const disablePrint = css`@media print { display: none; }`;
