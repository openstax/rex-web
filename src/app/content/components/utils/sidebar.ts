import { connect } from 'react-redux';
import { FlattenSimpleInterpolation } from 'styled-components';
import { css } from 'styled-components/macro';
import theme from '../../../theme';
import { AppState } from '../../../types';
import * as searchSelectors from '../../search/selectors';
import * as contentSelectors from '../../selectors';
import { State } from '../../types';

export const isVerticalNavOpenConnector = connect((state: AppState) => ({
  isDesktopSearchOpen: !!searchSelectors.query(state),
  isTocOpen: contentSelectors.tocOpen(state),
  isVerticalNavOpen: searchSelectors.searchResultsOpen(state) || contentSelectors.tocOpen(state),
}));

export const styleWhenTocClosed = (closedStyle: FlattenSimpleInterpolation) => css`
  ${(props: {isTocOpen: State['tocOpen']}) =>
    props.isTocOpen === null && theme.breakpoints.mobile(closedStyle)}
  ${(props: {isTocOpen: State['tocOpen']}) =>
    props.isTocOpen === false && closedStyle}
`;

export const styleWhenSidebarClosed = (closedStyle: FlattenSimpleInterpolation) => css`
  ${(props: {isVerticalNavOpen: State['tocOpen']}) =>
    props.isVerticalNavOpen === null && theme.breakpoints.mobile(closedStyle)}
  ${(props: {isDesktopSearchOpen: boolean; isVerticalNavOpen: State['tocOpen']}) =>
    props.isDesktopSearchOpen === false && props.isVerticalNavOpen === false && closedStyle}
`;
