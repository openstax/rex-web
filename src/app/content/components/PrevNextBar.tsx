import React from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { connect } from 'react-redux';
import styled, { css } from 'styled-components/macro';
import { ChevronLeft } from 'styled-icons/boxicons-regular/ChevronLeft';
import { ChevronRight } from 'styled-icons/boxicons-regular/ChevronRight';
import { decoratedLinkStyle, textRegularLineHeight, textRegularStyle } from '../../components/Typography';
import theme from '../../theme';
import { AppState } from '../../types';
import * as select from '../selectors';
import { ArchiveTreeSection, Book } from '../types';
import { contentTextWidth } from './constants';
import ContentLink from './ContentLink';
import { disablePrint } from './utils/disablePrint';

const prevNextIconStyles = css`
  height: ${textRegularLineHeight}rem;
  width: ${textRegularLineHeight}rem;
`;

// tslint:disable-next-line:variable-name
const LeftArrow = styled(ChevronLeft)`
  ${prevNextIconStyles}
`;

// tslint:disable-next-line:variable-name
const RightArrow = styled(ChevronRight)`
  ${prevNextIconStyles}
  margin-top: 0.1rem;
`;

interface HidingContentLinkProps {
  book?: Book;
  page?: ArchiveTreeSection;
  side: 'left' | 'right';
}
// tslint:disable-next-line:variable-name
const HidingContentLinkComponent = ({page, book, side, ...props}: HidingContentLinkProps) =>
  page !== undefined && book !== undefined
    ? <ContentLink book={book} page={page} {...props} />
    : <span aria-hidden />;

// tslint:disable-next-line:variable-name
const HidingContentLink = styled(HidingContentLinkComponent)`
  ${decoratedLinkStyle}
  ${(props) => props.side === 'left' && theme.breakpoints.mobile(css`
    margin-left: -0.8rem;
  `)}
  ${(props) => props.side === 'right' && theme.breakpoints.mobile(css`
    margin-right: -0.8rem;
  `)}
`;

// tslint:disable-next-line:variable-name
const BarWrapper = styled.div`
  ${disablePrint}
  ${textRegularStyle}
  overflow: visible;
  width: 100%;
  max-width: ${contentTextWidth}rem;
  justify-content: space-between;
  height: 4rem;
  display: flex;
  align-items: center;
  margin: 5rem auto 3rem auto;
  border-top: solid 0.1rem ${theme.color.neutral.darkest};
  border-bottom: solid 0.1rem ${theme.color.neutral.darkest};

  a {
    border: none;
    display: flex;
  }

  ${theme.breakpoints.mobile(css`
    margin: 3.5rem auto 3rem auto;
    border: 0;
  `)}
`;

interface PropTypes {
  book?: Book;
  prevNext: null | {
    prev?: ArchiveTreeSection;
    next?: ArchiveTreeSection;
  };
}

// tslint:disable-next-line:variable-name
const PrevNextBar = ({book, prevNext}: PropTypes) => {
  const { formatMessage } = useIntl();

  if (!prevNext) {
    return null;
  }

  return <BarWrapper data-analytics-region='prev-next'>
    <HidingContentLink side='left'
      book={book}
      page={prevNext.prev}
      aria-label={formatMessage({id: 'i18n:prevnext:prev:aria-label'})}
      data-analytics-label='prev'
    >
      <LeftArrow />
      <FormattedMessage id='i18n:prevnext:prev:text'>
        {(msg) => msg}
      </FormattedMessage>
    </HidingContentLink>

    <HidingContentLink side='right'
      book={book}
      page={prevNext.next}
      aria-label={formatMessage({id: 'i18n:prevnext:next:aria-label'})}
      data-analytics-label='next'
    >
      <FormattedMessage id='i18n:prevnext:next:text'>
        {(msg) => msg}
      </FormattedMessage>
      <RightArrow />
    </HidingContentLink>
  </BarWrapper>;
};

export default connect(
  (state: AppState) => ({
    book: select.book(state),
    prevNext: select.prevNextPage(state),
  })
)(PrevNextBar);
