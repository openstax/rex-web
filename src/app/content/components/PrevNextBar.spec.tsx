import React from 'react';
import renderer from 'react-test-renderer';
import createTestStore from '../../../test/createTestStore';
import { book as archiveBook, lastPage, page as firstPage, shortPage } from '../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../test/mocks/osWebLoader';
import TestContainer from '../../../test/TestContainer';
import { AppState, Store } from '../../types';
import { receiveBook, receivePage } from '../actions';
import { initialState } from '../reducer';
import { formatBookData } from '../utils';
import PrevNextBar from './PrevNextBar';

jest.mock('../../../config.books', () => {
  const mockBook = (jest as any).requireActual('../../../test/mocks/archiveLoader').book;
  return { [mockBook.id]: { defaultVersion: mockBook.version } };
});

const book = formatBookData(archiveBook, mockCmsBook);

describe('PrevNextBar', () => {
  let store: Store;

  beforeEach(() => {
    const state = {
      content: initialState,
    } as any as AppState;
    store = createTestStore(state);
  });

  const render = () => <TestContainer store={store}>
    <PrevNextBar />
  </TestContainer>;

  it('renders `null` with no page or book', () => {
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when you pass a page and book', () => {
    store.dispatch(receivePage({...shortPage, references: []}));
    store.dispatch(receiveBook(book));
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders `null` when passed a page that isn\'t in the book tree', () => {
    store.dispatch(receivePage({...shortPage, id: 'asdfasdfasd', references: []}));
    store.dispatch(receiveBook(book));
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders only `next` element correctly', () => {
    store.dispatch(receivePage({...firstPage, references: []}));
    store.dispatch(receiveBook(book));
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders only `prev` element correctly', () => {
    store.dispatch(receivePage({...lastPage, references: []}));
    store.dispatch(receiveBook(book));
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
