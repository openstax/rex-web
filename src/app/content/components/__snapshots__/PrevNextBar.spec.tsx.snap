// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrevNextBar renders \`null\` when passed a page that isn't in the book tree 1`] = `
<div
  className="PrevNextBar__BarWrapper-sc-13m2i12-3 fEZPiF"
  data-analytics-region="prev-next"
>
  <span
    aria-hidden={true}
  />
  <a
    aria-label="Next Page"
    className="PrevNextBar__HidingContentLink-sc-13m2i12-2 TdchT"
    data-analytics-label="next"
    href="books/book-slug-1/pages/test-page-1"
    onClick={[Function]}
  >
    Next
    <svg
      aria-hidden="true"
      className="PrevNextBar__RightArrow-sc-13m2i12-1 hAIJlW StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10.707 17.707L16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
      />
    </svg>
  </a>
</div>
`;

exports[`PrevNextBar renders \`null\` with no page or book 1`] = `null`;

exports[`PrevNextBar renders correctly when you pass a page and book 1`] = `
<div
  className="PrevNextBar__BarWrapper-sc-13m2i12-3 fEZPiF"
  data-analytics-region="prev-next"
>
  <a
    aria-label="Previous Page"
    className="PrevNextBar__HidingContentLink-sc-13m2i12-2 eMxlJc"
    data-analytics-label="prev"
    href="books/book-slug-1/pages/2-test-page-3"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="PrevNextBar__LeftArrow-sc-13m2i12-0 bBoLai StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13.293 6.293L7.586 12l5.707 5.707 1.414-1.414L10.414 12l4.293-4.293z"
      />
    </svg>
    Previous
  </a>
  <a
    aria-label="Next Page"
    className="PrevNextBar__HidingContentLink-sc-13m2i12-2 TdchT"
    data-analytics-label="next"
    href="books/book-slug-1/pages/4-test-page-5"
    onClick={[Function]}
  >
    Next
    <svg
      aria-hidden="true"
      className="PrevNextBar__RightArrow-sc-13m2i12-1 hAIJlW StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10.707 17.707L16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
      />
    </svg>
  </a>
</div>
`;

exports[`PrevNextBar renders only \`next\` element correctly 1`] = `
<div
  className="PrevNextBar__BarWrapper-sc-13m2i12-3 fEZPiF"
  data-analytics-region="prev-next"
>
  <span
    aria-hidden={true}
  />
  <a
    aria-label="Next Page"
    className="PrevNextBar__HidingContentLink-sc-13m2i12-2 TdchT"
    data-analytics-label="next"
    href="books/book-slug-1/pages/1-introduction-to-science-and-the-realm-of-physics-physical-quantities-and-units"
    onClick={[Function]}
  >
    Next
    <svg
      aria-hidden="true"
      className="PrevNextBar__RightArrow-sc-13m2i12-1 hAIJlW StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10.707 17.707L16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
      />
    </svg>
  </a>
</div>
`;

exports[`PrevNextBar renders only \`prev\` element correctly 1`] = `
<div
  className="PrevNextBar__BarWrapper-sc-13m2i12-3 fEZPiF"
  data-analytics-region="prev-next"
>
  <a
    aria-label="Previous Page"
    className="PrevNextBar__HidingContentLink-sc-13m2i12-2 eMxlJc"
    data-analytics-label="prev"
    href="books/book-slug-1/pages/d-glossary-of-key-symbols-and-notation"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="PrevNextBar__LeftArrow-sc-13m2i12-0 bBoLai StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13.293 6.293L7.586 12l5.707 5.707 1.414-1.414L10.414 12l4.293-4.293z"
      />
    </svg>
    Previous
  </a>
  <span
    aria-hidden={true}
  />
</div>
`;
